src/routes/User.js

```diff
@@ -0,0 +1,82 @@
+import React, {useState, useEffect} from 'react';
+import axios from 'axios';
+import styled from 'styled-components';
+import Cookies from 'js-cookie';
+
+import Candidate from '../components/Candidate';
+
+function User({ history }) {
+  const candidatesUrl = 'http://ec2-3-34-5-220.ap-northeast-2.compute.amazonaws.com:8080/candidates';
+  const voteUrl = `http://ec2-3-34-5-220.ap-northeast-2.compute.amazonaws.com:8080/vote?id=`;
+  const [candidates, setCandidates] = useState([]);
+  const [rerenderTrigger, setRerenderTrigger] = useState(true);
+  let candidatesData = [];
+  let sortedCandidates = [];
+
+  useEffect(() => {
+    const getCandidate = async () => {
+      try {
+        const response = await axios.get(candidatesUrl);
+        candidatesData = response.data;
+      } catch (e) {
+        console.log('Candidates error : ', e);
+      }
+    }
+    
+    async function sortVotes() {
+      sortedCandidates = candidatesData
+        .sort((a, b) => {return b.voteCount - a.voteCount})
+    }
+
+    async function getAndSort() {
+      await getCandidate();
+      await sortVotes();
+      setCandidates(sortedCandidates);
+    }
+    getAndSort();
+  }, [rerenderTrigger]);
+
+  async function vote(cid, name) {
+    try {
+      await axios.get(voteUrl+cid);
+      await alert(`${name}님에게 투표 완료 !!!`);
+      await setRerenderTrigger(rerenderTrigger ? false : true);
```

Comment:
이렇게 해봐도 좋을 것 같아요

```suggestion
      await setRerenderTrigger(!rerenderTrigger);
```

다만, rerenderTrigger보다 window.location.reload()를 이용해서 새로고침해보는 것도 나쁘지 않아보이네요

Reply:
> 이렇게 해봐도 좋을 것 같아요
> 
> 다만, rerenderTrigger보다 window.location.reload()를 이용해서 새로고침해보는 것도 나쁘지 않아보이네요

새로고침 문제가 종종 생기는데..! reload()로 해결 가능하면 대박